src/esrs/state.rs

```diff
@@ -1,17 +1,43 @@
 use uuid::Uuid;
 
+use crate::esrs::store::EventStoreLockGuard;
 use crate::types::SequenceNumber;
 
 /// The internal state for an Aggregate.
-/// It contains and id representing the aggregate id, an incremental sequence number and a state
-/// defined by the user of this library.
-#[derive(Clone, Debug)]
+/// It contains:
+/// - an id uniquely representing the aggregate,
+/// - an incremental sequence number,
+/// - a lock representing the atomicity of the access to the aggregate,
+/// - a state defined by the user of this library.
 pub struct AggregateState<S> {
     pub(crate) id: Uuid,
     pub(crate) sequence_number: SequenceNumber,
+    pub(crate) lock: Option<EventStoreLockGuard>,
     pub(crate) inner: S,
 }
 
+impl<S: Clone> Clone for AggregateState<S> {
+    fn clone(&self) -> Self {
+        Self {
+            id: self.id,
+            sequence_number: self.sequence_number,
+            inner: self.inner.clone(),
+            lock: None,
+        }
+    }
+}
+
+impl<S: std::fmt::Debug> std::fmt::Debug for AggregateState<S> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("AggregateState")
+            .field("id", &self.id)
+            .field("sequence_number", &self.sequence_number)
+            .field("inner", &self.inner)
+            .field("lock", &self.lock.is_some())
+            .finish()
+    }
+}
```

Comment:
Manual implementation of `Debug` as it could not be derived automatically anymore. I just went with `lock` as boolean as the internals are not really important; could have simply skipped the field instead but at this least this might give some insight.